reframe/core/pipeline.py

```diff
@@ -1038,7 +1041,7 @@ def _setup_perf_logging(self):
 
     @_run_hooks()
     @final
-    def setup(self, partition, environ, **job_opts):
+    def setup(self, partition, environ, sched=None, **job_opts):
```

Comment:
This is a breaking change of the interface, despite giving it a default value. Users override `setup()` with the old signature, and call `super().setup(partition, environ, **job_opts)`, so your `sched` argument will be missed. You should sneak it instead in `job_opts` passing it as a "special" reframe argument, e.g., `_rfm_sched_type`. Then you look for that in `job_opts` inside `setup()`.